sycl/doc/extensions/experimental/sycl_ext_codeplay_kernel_fusion.asciidoc

```diff
@@ -0,0 +1,678 @@
+= sycl_ext_codeplay_kernel_fusion
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Codeplay Software Limited.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 5 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+[NOTE]
+==== 
+This is an experimental extension for the SYCL specification.
+Exceptions while in fusion mode can leave a `queue` in an unknown fusion state,
+as no RAII-based management of fusion is done. Passing a `queue` in fusion mode
+to third-party libraries can make assumptions about the kernels enqueued by the
+library that might change over time. 
+
+This experimental proposal is intended to collect experience and early feedback
+on an API for kernel fusion in SYCL to inform a future extension proposal
+addressing the mentioned problems. 
+====
+
+[NOTE]
+====
+This extension is currently being implemented in {dpcpp} only for kernels in
+SPIRV format. Attempting to use this extension in kernels using a different
+format will result in an error, following this extension's design.
+====
```

Comment:
This note seems out-of-date to me.  The API semantics described below do not report an error when kernels cannot be fused.  Instead, they just silently execute without fusion.

Reply:
Yes, there is no error, fusion is simply aborted (with a warning message if `SYCL_RT_WARNING_LEVEL` >= 1). We updated the note in the latest commit.